tuf/api/metadata.py

```diff
@@ -1447,28 +1603,40 @@ class Delegations:
             defines which keys are required to sign the metadata for a specific
             role. The roles order also defines the order that role delegations
             are considered during target searches.
+        succinct_roles: Contains succinct information about hash bin
+            delegations. For more information see: https://github.com/theupdateframework/taps/blob/master/tap15.md
         unrecognized_fields: Dictionary of all attributes that are not managed
             by TUF Metadata API
 
+    Exactly one of ``roles`` and ``succinct_roles`` must be set.
+
+    Note: succinct_roles is a feature not yet part of the specification.
```

Comment:
API documentation should be direct: why are we telling this to the user of this API, and what can they do about it? Something like 

```suggestion
    Note that succinct roles is not a TUF specification feature yet and setting
    `succinct_roles` to a value makes the resulting metadata non-compliant.
    The metadata will not be accepted as valid by specification compliant 
    clients such as those built with python-tuf <= 1.1.0.
```

Could even move this to the actual argument description :shrug: